airflow/providers/amazon/aws/links/base.py

```diff
@@ -0,0 +1,87 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from datetime import datetime
+from typing import TYPE_CHECKING, ClassVar, Optional
+
+from airflow.models import BaseOperatorLink, XCom
+
+if TYPE_CHECKING:
+    from airflow.models import BaseOperator
+    from airflow.models.taskinstance import TaskInstanceKey
+    from airflow.utils.context import Context
+
+
+BASE_AWS_CONSOLE_LINK = "https://console.{aws_domain}"
+
+
+class BaseAwsLink(BaseOperatorLink):
+    """Base Helper class for constructing AWS Console Link"""
+
+    name: ClassVar[str]
+    key: ClassVar[str]
+    format_str: ClassVar[str]
+
+    @staticmethod
+    def get_aws_domain(aws_partition) -> Optional[str]:
+        if aws_partition == "aws":
+            return "aws.amazon.com"
+        elif aws_partition == "aws-cn":
+            return "amazonaws.cn"
+        elif aws_partition == "aws-us-gov":
+            return "amazonaws-us-gov.com"
+
+        return None
+
+    def get_link(
+        self,
+        operator,
+        dttm: Optional[datetime] = None,
+        ti_key: Optional["TaskInstanceKey"] = None,
+    ) -> str:
+        if ti_key is not None:
```

Comment:
```suggestion
        if ti_key:
```

Reply:
No way to change it right now

```
Check that providers are 2.1 compatible................................................Failed
- hook id: check-airflow-2-1-compatibility
- exit code: 1

Found Airflow 2.2 compatibility problems in providers:

In airflow/providers/amazon/aws/links/base.py:65 there is a forbidden construct (Airflow 2.3.0 only):

        if ti_key:
            conf = XCom.get_value(key=self.key, ti_key=ti_key)

When you use XCom.get_value( in providers, it should be in the form:

if ti_key is not None:
    value = XCom.get_value(...., ti_key=ti_key)

See: https://airflow.apache.org/docs/apache-airflow-providers/howto/create-update-providers.html#using-providers-with-dynamic-task-mapping
```

Reply:
Huh.   Alright, carry on then.   Sorry about that.